mitmproxy/addons/strip_dns_https_records.py

```diff
@@ -0,0 +1,35 @@
+from mitmproxy import ctx
+from mitmproxy import dns
+from mitmproxy.net.dns import types
+
+
+class StripDnsHttpsRecords:
+    def load(self, loader):
+        loader.add_option(
+            "strip_ech",
+            bool,
+            True,
+            "Strip Encrypted ClientHello (ECH) data from DNS HTTPS records so that mitmproxy can generate matching certificates.",
+        )
+
+    def dns_response(self, flow: dns.DNSFlow):
+        assert flow.response
+        if ctx.options.strip_ech:
+            for answer in flow.response.answers:
+                if answer.type == types.HTTPS:
+                    answer.https_ech = None
+        if not ctx.options.http3:
+            for answer in flow.response.answers:
+                if (
+                    answer.type == types.HTTPS
+                    and answer.https_alpn is not None
+                    and any(
+                        a == b"h3" or a.startswith(b"h3-") for a in answer.https_alpn
```

Comment:
```suggestion
                        a.startswith(b"h3") for a in answer.https_alpn
```
can we do this? (and similar change few lines below)

Reply:
We surely can, but that makes the code less obvious I think. :) Right now it's more obvious that we want `h3` or `h3-` draft variants. With just `.startswith(b"h3")`, you may quickly be wondering why it's not just `==`.

(the best solution here is a comment, which I will add now)